cloud/pkg/controller/controller/downstream.go

```diff
@@ -189,8 +193,31 @@ func (dc *DownstreamController) syncEdgeNodes(stop chan struct{}) {
 			switch e.Type {
 			case watch.Added:
 				dc.lc.UpdateEdgeNode(node.ObjectMeta.Name)
+				fallthrough
 			case watch.Modified:
-				continue
+				// TODO: keep node cache for edge node to check if status has changed to Running from not running
```

Comment:
The current cache of edgeNodes is just the name. I will update it to save status for comparison